### DISCUSSION TITLE
Connecting the SD card to the ESP32 SC01-plus

### DISCUSSION THREAD

**Discussion by powersoft** [sukesh-ak/ESP32-TUX#20] [Q&A] [answered]
Hello, I have an sc01-plus, and it is working fine. The only thing I can't get running is the sdcard. I use the Arduino envoirment to programm. Do you have tips for me to get it running.

This is the program I have written to test it.

Thanks for any help

`#define LGFX_USE_V1         // set to use new version of library
#include <LovyanGFX.hpp>    // main library
#include "config.h"

#include "FS.h"       // File System header
#include "SD.h"       // SD Card header 

static LGFX lcd;            // declare display variable

#define sdSelectPin 41
void setup() {
  Serial.begin(500000);
  lcd.init();
  lcd.setRotation(1); 
  lcd.clear(TFT_RED);
  if (!SD.begin(sdSelectPin))
  {
    Serial.println("SD Card initialisation failed!");
  }
  Serial.println("Ready");
}
void loop() {
  
}

**Comment by sukesh-ak** [comment, marked as answer]
> Hello, I have an sc01-plus, and it is working fine. The only thing I can't get running is the sdcard. I use the Arduino envoirment to programm. Do you have tips for me to get it running.
> 
> This is the program I have written to test it.
> 
> Thanks for any help
> 
> `#define LGFX_USE_V1 // set to use new version of library #include <LovyanGFX.hpp> // main library #include "config.h"
> 
> #include "FS.h" // File System header #include "SD.h" // SD Card header
> 
> static LGFX lcd; // declare display variable
> 
> #define sdSelectPin 41 void setup() { Serial.begin(500000); lcd.init(); lcd.setRotation(1); lcd.clear(TFT_RED); if (!SD.begin(sdSelectPin)) { Serial.println("SD Card initialisation failed!"); } Serial.println("Ready"); } void loop() {
> 
> }

Try using `SD2.begin` instead of `SD.begin` and see if it works since this board I believe has the SD card on a different SPI bus.
